packages/design-system/src/components/InlineEditing/Primitives/InlineEditingPrimitive.tsx

```diff
@@ -57,22 +79,37 @@ const InlineEditingPrimitive = forwardRef(
 			loading,
 			ariaLabel,
 			mode,
-			onEdit = () => {},
 			onCancel = () => {},
 			onToggle = () => {},
 			required = false,
-			defaultValue,
 			label,
 			hasError,
 			description,
 			placeholder,
 			renderValueAs,
+			defaultValue,
+			onEdit = () => {},
+			value,
+			onChangeValue,
 			...rest
 		} = props;
+
 		const { t } = useTranslation(I18N_DOMAIN_DESIGN_SYSTEM);
+
 		const [isEditing, setEditing] = useState<boolean>(false);
-		const [value, setValue] = useState<string | undefined>(defaultValue);
+
+		// Both modes (controlled & uncontrolled) - store currently typed value
+		const [inProgressValue, setInProgressValue] = useState<string | undefined>(defaultValue);
+
+		// Both modes (controlled & uncontrolled) - store a snapshot value
+		const [snapshotValue, setSnapshotValue] = useState<string | undefined>(defaultValue);
+
 		const toggleEditionMode = (isEditionMode: boolean) => {
+			if (isEditionMode) {
+				// Both modes - initiate inProgressValue with stored value
+				setInProgressValue(snapshotValue);
+			}
```

Comment:
Isn't `inProgressValue` already to `defaultValue`

Reply:
It should, as `defaultValue` is the default value of the state 🤔 
Maybe it is for successive toggle of the edit mode ?

Reply:
regardless your opinion about controlled mode / uncontrolled mode here, defaultValue is designed for uncontrolled mode and is not used when user provider value/onChangeValue